### PR TITLE
Adding Block Entropy API endpoints for chat, image, and video generation

### PR DESCRIPTION
**Pull request by edk208**
## Checklist:

- [ x ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

_What is the reason for a change?_
I am the creator of block entropy, an open source generative AI platform.  In this PR, I am enabling the use of the BE endpoints for LLM chat completion usage, as well as the image generation endpoints in the extensions.  

_What did you do to achieve this?_
I made changes to the index.html to add the data source, the openai.js for the chat completions, tokenizers.js to best match the type of tokenizer, and added to stable-diffusion.js and index.js for the image and video endpoints.  There are also some changes to fix linting isues.

_How would a reviewer test the change?_
Sign up for an account at blockentropy.ai, limited funds are available upon new account creation.  Create an API key.  
In the Silly Tavern interface, select chat completion and choose Block entropy.  Enter the API key and select a model from the drop down window.  
For image creation, in the extensions tab, select image generation and select an image model from the drop down list.  For animation, the models with the prefix "be-anim" send back an animated gif (note that the sizes of animated gifs can be quite large, between 1-2.5M)

### CODE REVIEW

File: public/scripts/openai.js [hunk 1]
```diff
@@ -4302,6 +4330,19 @@ async function onModelChange() {
         oai_settings.temp_openai = Math.min(oai_max_temp, oai_settings.temp_openai);
         $('#temp_openai').attr('max', oai_max_temp).val(oai_settings.temp_openai).trigger('input');
     }
+    if (oai_settings.chat_completion_source === chat_completion_sources.BLOCKENTROPY) {
+        if (oai_settings.max_context_unlocked) {
+            $('#openai_max_context').attr('max', unlocked_max);
+        } else {
+            $('#openai_max_context').attr('max', max_32k);
```
Comment (Cohee1207) [inline public/scripts/openai.js]: Are you sure that all of your models only support 32k tokens of context? Should this information be available in the API?

File: public/scripts/extensions/stable-diffusion/settings.html [hunk 2]
```diff
@@ -48,6 +48,7 @@
                 <option value="auto">Stable Diffusion Web UI (AUTOMATIC1111)</option>
                 <option value="horde">Stable Horde</option>
                 <option value="togetherai">TogetherAI</option>
+                <option value="blockentropy">Block Entropy</option>
```
Comment (Cohee1207) [inline public/scripts/extensions/stable-diffusion/settings.html]: SD sources are sorted alphabetically. It should go first, before ComfyUI

File: public/scripts/openai.js (hunk 1 above)
Comment (edk208) [reply public/scripts/openai.js]: they do not.  i will adjust accordingly thanks
edit: adjusted based upon model

File: public/scripts/extensions/stable-diffusion/settings.html (hunk 2 above)
Comment (edk208) [reply public/scripts/extensions/stable-diffusion/settings.html]: ok will make that adjustment
edit: adjusted based upon alphabetical order

**Comment by edk208** [comment]
thanks for the feedback, adjusted as requested.

**Review by Cohee1207 — APPROVED** [review]
LGTM

**Pull request merged**